### PR TITLE
Issue #36564 update regex for spread operator

### LINKED ISSUE #36564: Spread operator only accepts array literal
**Describe your problem and how to reproduce it:**
In [this lesson on the spread operator](https://learn.freecodecamp.org/javascript-algorithms-and-data-structures/es6/use-the-spread-operator-to-evaluate-arrays-in-place/), I ran into the following issue.

This particular lesson has three tests:
1. arr2 is correct copy of arr1.
2. ... spread operator was used to duplicate arr1.
3. arr2 remains unchanged when arr1 is changed.

When I change the default text (`arr2 = []; // change this line`) to my solution of choice (`arr2 = Array(...arr1);`) I find that tests 1 and 3 pass, while test 2 fails. But, I'm clearly using the spread operator here.

I can get test 2 to pass by using an array literal, rather than a function (e.g. `arr2 = [...arr1];`), however, both solutions should be valid, as they both fulfill all requirements for this lesson.

**Add a Link to the page with the problem:**
https://learn.freecodecamp.org/javascript-algorithms-and-data-structures/es6/use-the-spread-operator-to-evaluate-arrays-in-place/

### PR DESCRIPTION
**Pull request by enrider**
- [x] I have read [freeCodeCamp's contribution guidelines](https://github.com/freeCodeCamp/freeCodeCamp/blob/master/CONTRIBUTING.md).
- [x] My pull request has a descriptive title (not a vague title like `Update index.md`)
- [x] My pull request targets the `master` branch of freeCodeCamp.
- [x] None of my changes are plagiarized from another source without proper attribution.
- [x] All the files I changed are in the same world language (for example: only English changes, or only Chinese changes, etc.)
- [x] My changes do not use shortened URLs or affiliate links.

Closes #36564

### CODE REVIEW

File: curriculum/challenges/english/02-javascript-algorithms-and-data-structures/es6/use-the-spread-operator-to-evaluate-arrays-in-place.english.md [hunk 1]
```diff
@@ -45,7 +45,7 @@ tests:
   - text: <code>arr2</code> should be correct copy of <code>arr1</code>.
     testString: assert(arr2.every((v, i) => v === arr1[i]));
   - text: <code>...</code> spread operator was used to duplicate <code>arr1</code>.
-    testString: assert(code.match(/\[\s*...arr1\s*\]/g));
+    testString: assert(code.match(/\s*...arr1\s*/g));
```
Comment (RandellDawson) [inline curriculum/challenges/english/02-javascript-algorithms-and-data-structures/es6/use-the-spread-operator-to-evaluate-arrays-in-place.english.md]: @enrider The problem with your regex is that it should not just be checking if the spread operator is used on `arr1`.  It is supposed to validate the spread operator correctly creates a copy of `arr1`.

If you read the documentation for challenge related PRs, you will find you must test the changes locally (`npm run test:curriculum`) to confirm you do not make any breaking changes.  Your change allows the user to not change any of the challenge seed code and still pass all of the tests.

Also, the original regex should not have used `...`.  Instead, it should have been `\.\.\.`.  I have made a suggestion which does pass all of the tests and correctly fails when no code is changed.  You can click `Commit suggestion` and it apply the suggested change on your behalf.

```suggestion
    testString: assert(code.match(/Array\(\s*\.\.\.arr1\s*\)|\[\s*\.\.\.arr1\s*\]/));
```

**Review by RandellDawson — CHANGES_REQUESTED** [review]
@enrider Thank you for you PR.  See my comments and requested change.

**Review by moT01 — APPROVED** [review]
LGTM! 🎉 

Want to approve and merge this @RandellDawson ?

**Review by RandellDawson — APPROVED** [review]

**Comment by RandellDawson** [comment]
Congrats @enrider  on your first pull request (PR)! 🎉

Thank you for your contribution to the page! 👍
We're happy to accept these changes, and look forward to future contributions. 📝

**Pull request merged**